test/parallel/test-child-process-stdio-overlapped.js

```diff
@@ -0,0 +1,79 @@
+// Test for "overlapped" stdio option. This test uses the "overlapped-checker"
+// helper program which basically a specialized echo program.
+//
+// The test has two goals:
+//
+// - Verify that overlapped I/O works on windows. The test program will deadlock
+//   if stdin doesn't have the FILE_FLAG_OVERLAPPED flag set on startup (see
+//   test/overlapped-checker/main_win.c for more details).
+// - Verify that "overlapped" stdio option works transparently as a pipe (on
+//   unix/windows)
+//
+// This is how the test works:
+//
+// - This script assumes only numeric strings are written to the test program
+//   stdout.
+// - The test program will be spawned with "overlapped" set on stdin and "pipe"
+//   set on stdout/stderr and at startup writes a number to its stdout
+// - When this script receives some data, it will parse the number, add 50 and
+//   write to the test program's stdin.
+// - The test program will then echo the number back to us which will repeat the
+//   cycle until the number reaches 200, at which point we send the "exit"
+//   string, which causes the test program to exit.
+// - Extra assertion: Every time the test program writes a string to its stdout,
+//   it will write the number of bytes written to stderr.
```

Comment:
Can you make it explicit here that the helper program is supposed to hang if overlapped I/O is somehow not working?

Reply:
Above there's the following comment:

```
// - Verify that overlapped I/O works on windows. The test program will deadlock
//   if stdin doesn't have the FILE_FLAG_OVERLAPPED flag set on startup (see
//   test/overlapped-checker/main_win.c for more details).
```

Is that enough?